src/harness/cbsd_to_sas.py

```diff
@@ -0,0 +1,271 @@
+        'ini': 'HEARTBEAT'},
+    'rel': {
+        'name': 'RELINQUISHMENT',
+        'path': '/relinquishment',
+        'req': 'relinquishmentRequest',
+        'res': 'relinquishmentResponse',
+        'ini': 'RELINQUISHMENT'},
+    'reset': {
+        'name': 'RESET',
+        'path': '/reset',
+        'req': 'resetRequest',
+        'res': 'resetResponse'}}
+
+
+# Classes
+
+class SasInterfaceImpl(SasInterface):
+    """ Class CBSD to SAS Gateway
+        Send/Receive messages to/from given URL
+    """
+
+    def __init__(self):
+
+        # Logging Instance creation
+        logger = logging.getLogger(__name__)
+        logger.setLevel(logging.DEBUG)
+        # Handle set-up
+        handler = logging.FileHandler(LOG_FILE)
+        handler.setLevel(logging.DEBUG)
+        # Couple handler to logging instance
+        logger.addHandler(handler)
+
+        super(SasInterfaceImpl, self).__init__(logger, handler)
+
+        self.logger = logger
+        self.handler = handler
+
+        # Prep SAS to SAS Configuration file set-up section/options
+        self.LoadSettings()
+
+        self.handler.setFormatter(LOG_FORMAT)
+        self.logger.debug('*** SasInterfaceImpl ****')
+
+    def GetIniData(self, option, section):
+        """ Return the configuration file defined option value """
+        self.handler.setFormatter(LOG_FORMAT)
+
+        try:
+            return(self.ini[section][option])
+
+        except Exception as e:
+            self.logger.error(
+                '[{}] {} Bad data retrieval'.format(section, option))
+            sys.exit(1)
+
+    def GetLocalTime(self):
+        """ Return the local machine time in UTC time zone """
+
+        try:
+            utc = timezone('UTC')
+            return(utc.localize(datetime.utcnow()))
+
+        except Exception as e:
+            self.logger.error('Time Zone error: {}'.format(e))
+            sys.exit(1)
+
+    # Message Requests
+
+    def RegistrationRequest(self, reg_obj):
+        """ Send Registration Request to SAS """
+
+        # Build the message
+        req_array = {}
+        if(not isinstance(reg_obj, list)):
+            reg_obj = [reg_obj]
+        req_array[RESOURCES['reg']['req']] = reg_obj
+
+        # Send request, get response
+        results = self.SendMessage(PUSH, RESOURCES['reg'], req_array)
+        return(results[RESOURCES['reg']['res']])
+
+    def DeregistrationRequest(self, reg_obj):
+        """ Send Deregistration Request to SAS """
+
+        # Initialize request array object
+        req_array = {}
+        if(not isinstance(reg_obj, list)):
+            reg_obj = [reg_obj]
+        req_array[RESOURCES['dereg']['req']] = [
+            {'cbsdId': r['cbsdId']} if 'cbsdId' in r else {} for r in reg_obj]
+
+        # Send request, get response
+        results = self.SendMessage(PUSH, RESOURCES['dereg'], req_array)
+        return(results[RESOURCES['dereg']['res']])
+
+    def SpectrumInquiryRequest(self, reg_obj, spec_obj):
```

Comment:
Also, in this way, we can add a sas_to_sas.py module for the SAS to SAS testing, reusing the sas_interface module